src/app/quizPage.js

```diff
@@ -0,0 +1,222 @@
+    const li = getTemplateContent(liTemplate)[0];
+    const liFirstElem = li.children[0]
+
+    if (mode.answerType === "image") {
+        // first child of li receives an image
+        liFirstElem.classList.add(QUIZ_PAGE_STYLES.quizAnswerImageClass)
+        const imgElem = createImgElement(answer) // get img url
+        liFirstElem.appendChild(imgElem)
+
+    } else if (mode.answerType === "text") {
+        // first child of li receives text
+        liFirstElem.classList.add(QUIZ_PAGE_STYLES.quizAnswerTextClass)
+        liFirstElem.innerText = answer // add question as an inner text
+    }
+    return li
+}
+
+// Updates the question number ona quiz page
+const updateQuestionCounter = (counterElem, questionNum) => {
+    counterElem.innerText = String(questionNum).padStart(2, '0'); //TODO maybe - total num of question in a game mode
+}
+
+// fires up on mouse up event on answers list li, additionally accepts questionSet
+// check which answer was selected
+// eventHandler is element to which eventListener was attached to
+function selectAnswer(questionSet, eventHandler) {
+    const answer = getAnswerFromElement(eventHandler);
+    if (answer) {
+        questionSet.correctAnswer.name === answer ? correctAnswerSelected(eventHandler) : wrongAnswerSelected(eventHandler);
+    } else {
+        throw new Error('Answer was not found')
+    }
+}
+
+
+// Read selected answer value from clicked list item
+const getAnswerFromElement = (target) => {
+    const targetClasses = [...target.classList];
+    let answer;
+    if (targetClasses.includes(QUIZ_PAGE_STYLES.uncheckedClass)) {
+        if (targetClasses.includes(QUIZ_PAGE_STYLES.quizAnswerTextClass)) {
+            answer = target.innerText
+        } else if (targetClasses.includes(QUIZ_PAGE_STYLES.quizAnswerImageClass)) {
+            answer = target.children[0].getAttribute("src")
+        }
+        return answer
+    }
+}
+
+const correctAnswerSelected = (selectedElem) => {
+    console.log("yes")
+    //TODO add correct-answer class and remove unchecked
+    selectedElem.classList.remove(QUIZ_PAGE_STYLES.uncheckedClass)
+    selectedElem.classList.add(QUIZ_PAGE_STYLES.correctAnswerClass)
+    //TODO store results
+    setTimeout(()=> {
+        resetQuizAfterQuestion();
+        renderNextQuestion(CURRENT_MODE);
+    }, TIMEOUT_AFTER_ANSWER_SELECTION)
+}
+
+const wrongAnswerSelected = (selectedElem) => {
+    console.log("no")
+    // add wrong-answer class and remove unchecked
+    selectedElem.classList.remove(QUIZ_PAGE_STYLES.uncheckedClass)
+    selectedElem.classList.add(QUIZ_PAGE_STYLES.wrongAnswerClass)
+    //TODO store results
+    setTimeout(()=> {
+        resetQuizAfterQuestion();
+        renderNextQuestion(CURRENT_MODE);
+    }, TIMEOUT_AFTER_ANSWER_SELECTION)
+}
+
+// removes question list items
+const resetQuizAfterQuestion = () => {
+    const quizBody = document.querySelector('#quiz-body');
+    const quizTemplate = document.getElementById('quiz-template');
```

Comment:
Isn't `querySelector('#quiz-body')` the same as `getElementById('quiz-body')`? I am not sure, just asking 😄

Reply:
yup, it would be the same, i could just use querySelector everywhere

Reply:
Well, I think it is better to use getElementById - because you know that you are searching for **specific id** and not for any other items. Internet says "it depends", see here:
https://stackoverflow.com/questions/26848289/javascript-queryselector-vs-getelementbyid